### DISCUSSION TITLE
What's the correct way to configure a StateProcessor to keep the deletion working via Doctrine

### DISCUSSION THREAD

**Discussion by herndlm** [api-platform/core#5107] [Q&A] [answered]
I did read [the docs](https://api-platform.com/docs/core/state-processors/#creating-a-custom-state-processor) of course and found also https://github.com/api-platform/core/pull/4859, but I realised that our processor, that decorates the Doctrine PersistProcessor, leads to deletion not working anymore because RemoveProcessor is not used dynamically any more, see https://github.com/api-platform/core/blob/v2.7.2/src/Doctrine/Orm/Metadata/Resource/DoctrineOrmResourceCollectionMetadataFactory.php#L110-L112

Based on the answer here I guess the docs should be updated to avoid breaking the DELETE operation as I did :)

**Comment by soyuka** [comment, marked as answer]
Decorate the remove processor and the persist processor, call either one based on your needs ? 

Would that work?

**Reply by herndlm** [reply]
yeah I think so. I just wanted to double-check basically. My plan is to decorate the PersistProcessor but internally inject both the PersistProcessor and RemoveProcessor and then use the one that is needed as the factory basically does. If you meant the same?

**Reply by herndlm** [reply]
maybe this should be done by default in the docs for Doctrine as well? I can provide a PR if you're interested

**Reply by herndlm** [reply]
Opened https://github.com/api-platform/docs/pull/1646, that makes clear what I meant.
